File: test-server-jdk8/src/main/java/org/newrelic/jmx/Cat.java
package org.newrelic.jmx;

import javax.management.MBeanRegistration;
import javax.management.MBeanServer;
import javax.management.ObjectName;

public class Cat implements CatMBean, MBeanRegistration {

  private String name;

  public Cat(String name) {
    this.name = name;
  }

  @Override
  public String getName() {
    return name;
  }

  public String toJSON() {
    return "{\"name\":\"" + name + "\"}";
  }

  @Override
  public String toString() {
    return toJSON();
  }

  @Override
  public ObjectName preRegister(MBeanServer server, ObjectName name) throws Exception {
    return new ObjectName("test:type=Cat,name=" + this.name);
  }

  @Override
  public void postRegister(Boolean registrationDone) {}

  @Override
  public void preDeregister() {}

  @Override
  public void postDeregister() {}
}

File: test-server/src/main/kotlin/org/newrelic/jmx/Cat.kt
package org.newrelic.jmx

import javax.management.MBeanRegistration
import javax.management.MBeanServer
import javax.management.ObjectName

interface CatMBean {
    val name: String
}

class Cat(override val name: String) : CatMBean, MBeanRegistration {
    override fun preDeregister() {}
    override fun postRegister(registrationDone: Boolean?) {}
    override fun postDeregister() {}

    override fun preRegister(server: MBeanServer?, oname: ObjectName?): ObjectName =
            ObjectName("test:type=Cat,name=$name")

    override fun toString(): String {
        return "Cat{name=$name}"
    }
}
